packages/blade/src/components/Input/BaseInput/BaseInputVisuals.tsx

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import type { ReactElement } from 'react';
+import type { BaseInputProps } from './BaseInput';
+import Box from '~components/Box';
+import type { BoxProps } from '~components/Box/Box';
+import { Text } from '~components/Typography';
+
+type InputVisuals = Pick<
+  BaseInputProps,
+  'leadingIcon' | 'prefix' | 'interactionElement' | 'suffix' | 'trailingIcon' | 'isDisabled'
+>;
+
+const getVisualContainerStyles = (): BoxProps => ({
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'center',
+});
+
+const getPrefixStyles = ({
+  hasLeadingIcon,
+  hasPrefix,
+}: {
+  hasLeadingIcon: boolean;
+  hasPrefix: boolean;
+}): BoxProps => {
+  if (hasPrefix && hasLeadingIcon) {
+    return {
+      paddingLeft: 'spacing.1',
+    };
+  }
+
+  if (hasPrefix && !hasLeadingIcon) {
+    return {
+      paddingLeft: 'spacing.3',
+    };
+  }
+
+  return {
+    paddingLeft: 'spacing.0',
+  };
+};
+
+const getInteractionElementStyles = ({
+  hasTrailingIcon,
+  hasInteractionElement,
+  hasSuffix,
+}: {
+  hasTrailingIcon: boolean;
+  hasInteractionElement: boolean;
+  hasSuffix: boolean;
+}): BoxProps => {
+  if (hasInteractionElement && (hasSuffix || hasTrailingIcon)) {
+    return {
+      paddingRight: 'spacing.1',
+    };
+  }
+
+  if (hasInteractionElement && !hasSuffix && !hasTrailingIcon) {
+    return {
+      paddingRight: 'spacing.3',
+    };
+  }
+
+  return {
+    paddingRight: 'spacing.0',
+  };
+};
+
+const getSuffixStyles = ({
+  hasTrailingIcon,
+  hasSuffix,
+}: {
+  hasTrailingIcon: boolean;
+  hasSuffix: boolean;
+}): BoxProps => {
+  if (hasSuffix && hasTrailingIcon) {
+    return {
+      paddingRight: 'spacing.1',
+    };
+  }
+
+  if (hasSuffix && !hasTrailingIcon) {
+    return {
+      paddingRight: 'spacing.3',
+    };
+  }
+
+  return {
+    paddingRight: 'spacing.0',
+  };
+};
+
+export const getInputVisualsToBeRendered = ({
```

Comment:
nit: 

```suggestion
export const getInputVisuals = ({
```

Reply:
literally had typed this and went back to `toBeRendered` primarily since we are not getting the actual Input Visuals we are just getting the info of what needs to be rendered.

Reply:
Oh just saw that, makes sense then.